### PR TITLE
Updating firestore.rules

### LINKED ISSUE #49: Review DB access rules

### PR DESCRIPTION
**Pull request by bocon13**
- Only authenticated user can create CLAs and addendums for themselves
- Users can only read CLAs and addendums that they have signed

TODO: still requires unit tests

closes #49

### CODE REVIEW

File: firestore.rules [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
-	return request.auth.token.email == resource.data.signer.value || isAdmin();
+// Checks to see if the signer of the existing CLA or Addendum is the authenticated user
+function didSign() {
+  return request.auth.token.email == resource.data.signer.value;
 }
 
+// Note: CLAs and Addendums cannot be modified or deleted
 service cloud.firestore {
   match /databases/{database}/documents {
     match /agreements/{cla} {
```
Comment (ccascone) [inline firestore.rules]:
```suggestion
    match /agreements/{agreementId} {
```

File: firestore.rules [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
+
+Today, only the signer of an agreement or addendum can read the document.
+
+1. In the future, we may allow office admins to amend agreements by adding addendums
+on behalf of a signer.
+
+If this happens, we will need to update the canAmendAgreements to check to see if
+the signer is authorized in the parent CLA as an admin.
+
+TODO: get() doesn't appear to work in a function. If that changes in the future,
```
Comment (ccascone) [inline firestore.rules]: since functions access the DB as admins, are rules evaluated? If not, then we should not worry about this.

Comment (bocon13) [reply firestore.rules]: Firebase admins (e.g. in functions) bypass the rules.

File: firestore.rules (hunk 1 above)
Comment (bocon13) [reply firestore.rules]: sure.. I don't think that parameter is being used at the moment anyway

**Comment by teone** [comment]
It would be great if firebase will tell you which rule (if not in the error
message at least somewhere in the console)

**Pull request merged**